specification/imds/data-plane/Microsoft.InstanceMetadataService/stable/2018-10-01/imds.json

```diff
@@ -0,0 +1,731 @@
+                        ],
+                        "description": "This must be set to 'true'.",
+                        "required": true
+                    },
+                    {
+                        "name": "resource",
+                        "in": "query",
+                        "type": "string",
+                        "description": "This is the urlencoded identifier URI of the sink resource for the requested Azure AD token. The resulting token contains the corresponding aud for this resource.",
+                        "required": true
+                    },
+                    {
+                        "name": "api-version",
+                        "in": "query",
+                        "type": "string",
+                        "enum": [
+                            "2018-02-01",
+                            "2018-04-02",
+                            "2018-10-01"
+                        ],
+                        "description": "This is the API version to use.",
+                        "required": true
+                    },
+                    {
+                        "name": "client_id",
+                        "in": "query",
+                        "type": "string",
+                        "description": "This identifies, by Azure AD client id, a specific explicit identity to use when authenticating to Azure AD. Mutually exclusive with object_id and msi_res_id.",
+                        "required": false
+                    },
+                    {
+                        "name": "object_id",
+                        "in": "query",
+                        "type": "string",
+                        "description": "This identifies, by Azure AD object id, a specific explicit identity to use when authenticating to Azure AD. Mutually exclusive with client_id and msi_res_id.",
+                        "required": false
+                    },
+                    {
+                        "name": "msi_res_id",
+                        "in": "query",
+                        "type": "string",
+                        "description": "This identifies, by urlencoded ARM resource id, a specific explicit identity to use when authenticating to Azure AD. Mutually exclusive with client_id and object_id.",
+                        "required": false
+                    },
+                    {
+                        "name": "authority",
+                        "in": "query",
+                        "type": "string",
+                        "description": "This indicates the authority to request AAD tokens from. Defaults to the known authority of the identity to be used.",
+                        "required": false
+                    },
+                    {
+                        "name": "bypass_cache",
+                        "in": "query",
+                        "type": "string",
+                        "enum": [
+                            "true"
+                        ],
```

Comment:
Please use the x-ms-enum extension here. Using the x-ms-enum extension when the enum only has one value and the parameter/property is required has a different meaning to it. More info over [here](https://github.com/Azure/autorest/tree/master/docs/extensions#single-value-enum-as-a-constant).